Sources/Models/RequestModel.swift

```diff
@@ -11,24 +11,69 @@ import UIKit
 open class RequestModel: Codable {
     public let id: String
     public let url: String
+    public let host: String?
+    public let port: Int?
+    public let scheme: String?
     public let date: Date
     public let method: String
-    public let headers: [String: String]?
+    public let headers: [String: String]
+    public var credentials: [String : String]
+    public var cookies: String?
     open var httpBody: Data?
     open var code: Int
     open var responseHeaders: [String: String]?
     open var dataResponse: Data?
     open var errorClientDescription: String?
     open var duration: Double?
     
-    init(request: NSURLRequest) {
+    init(request: NSURLRequest, session: URLSession?) {
         id = UUID().uuidString
         url = request.url?.absoluteString ?? ""
+        host = request.url?.host
+        port = request.url?.port
+        scheme = request.url?.scheme
         date = Date()
-        method = request.httpMethod ?? "N/A"
-        headers = request.allHTTPHeaderFields
+        method = request.httpMethod ?? "GET"
+        credentials = [:]
+        var headers = request.allHTTPHeaderFields ?? [:]
         httpBody = request.httpBody
         code = 0
+        
+        session?.configuration.httpAdditionalHeaders?.filter {  $0.0 != AnyHashable("Cookie") }
```

Comment:
I added this filter to avoid collecting duplicate headers here and below, at the end of the same method. 

Strictly speaking cookies are special headers that are often treated apart from other headers in several request representation. 
For instance: cURL uses the "-b" parameter to specify cookies instead of "-H"; Postman uses a "cookie" array in a JSON collection.

There are two scenarios that I have to verify/handle before confirm changes I made to this file:
- session.configuration.httpShouldSetCookies == true -> cookies are in a cookie jar handled by session (session.configuration.httpCookieStorage)
- session.configuration.httpShouldSetCookies == false -> the session configuration does not handle cookies because user may send them using other ways (see https://developer.apple.com/documentation/foundation/nsurlsessionconfiguration/1411589-httpshouldsetcookies?language=objc). If the user choose to handle cookies by himself/herself via httpAdditionalHeaders, we need to pick the cookie from this dictionary and fill the cookies property of this struct.

In both cases,  _headers_ dictionary must not include the "Cookie" header key/value.

Reply:
It sounds good! Since in a month no one remembers this, can you add a comment in the code about this? 🙇‍♂️